### PR TITLE
feat: add retry count parameter to function uploads

### PR DESCRIPTION
**Pull request by eduardoboucas**
#### Summary

Adds a retry count parameter to the `uploadDeployFunction` API call on retries.

Part of https://github.com/netlify/pod-compute/issues/140.

### CODE REVIEW

File: src/utils/deploy/upload-files.js [hunk 1]
```diff
@@ -79,10 +85,12 @@ const retryUpload = (uploadFn, maxRetry) =>
 
     const tryUpload = async () => {
       try {
-        const results = await uploadFn()
+        const results = await uploadFn(retryCount)
         return resolve(results)
```
Comment (danez) [inline src/utils/deploy/upload-files.js:89]: [dust] new line before return

File: src/utils/deploy/upload-files.js [hunk 2]
```diff
@@ -79,10 +85,12 @@ const retryUpload = (uploadFn, maxRetry) =>
 
     const tryUpload = async () => {
```
Comment (danez) [inline src/utils/deploy/upload-files.js:86]: The first argument is a counter, see https://github.com/MathieuTurcotte/node-backoff#event-ready could we use that? Only thing is that I think by reading the docs it starts at 0 for the first backoff, so we would need to do +1.

```suggestion
    const tryUpload = async (retryCount = 0) => {
```

**Comment by danez** [comment]
Also a test would be nice.

File: src/utils/deploy/upload-files.js (hunk 2 above)
Comment (eduardoboucas) [reply src/utils/deploy/upload-files.js:86]: Hmm maybe I'm missing something, but I'm not seeing any parameters being passed to the function.

File: src/utils/deploy/upload-files.js (hunk 1 above)
Comment (eduardoboucas) [reply src/utils/deploy/upload-files.js:90]: Tasting my own poison! 🙈 

Done in 400da760aa283649d2145643f0cfcb11a0b4876c.

File: src/utils/deploy/upload-files.js (hunk 2 above)
Comment (danez) [reply src/utils/deploy/upload-files.js:86]: https://github.com/MathieuTurcotte/node-backoff/blob/master/lib/backoff.js#L53 The event is passed some params, and as we use this function for the event we could use them.

**Comment by eduardoboucas** [comment]
> chore: add test

Done in https://github.com/netlify/cli/pull/4841/commits/6bec2f54656cef01b1121b4a8e34acd76c12f193.

File: src/utils/deploy/upload-files.js (hunk 2 above)
Comment (eduardoboucas) [reply src/utils/deploy/upload-files.js:86]: Right, but I'm logging the actual parameters that are passed to `tryUpload` and nothing comes up.

Comment (eduardoboucas) [reply src/utils/deploy/upload-files.js:86]: Ah, my bad. I was looking in the wrong place. Will update!

Comment (eduardoboucas) [reply src/utils/deploy/upload-files.js]: Done in https://github.com/netlify/cli/pull/4841/commits/6fed5241e61fa5f47903e7f25d1cca1ea1fa4e4b.

**Review by danez — APPROVED** [review]
🚀

**Pull request merged**